core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1222,6 +1293,12 @@ public void onSettingsChanged() {
         mergeScheduler.refreshConfig();
         // config().isEnableGcDeletes() or config.getGcDeletesInMillis() may have changed:
         maybePruneDeletedTombstones();
+        if (engineConfig.getOptimizeAutoGeneratedIds() == false) {
+            // this is an anti-viral settings you can only opt out for the entire index
```

Comment:
I think I understand what your are saying - there is no way to re-enable this on a live engine - but it will be good to explicitly say that rather than "anti-viral" (if I get you correctly :))